src/System.Data.SqlClient/tests/ManualTests/SQL/AsyncTest/BeginExecReaderAsyncTest.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Data.SqlClient.ManualTesting.Tests
+{
+    public static class BeginExecReaderAsyncTest
+    {
+        private static string GenerateCommandText()
+        {
+            int suffix = (new Random()).Next(5000);
+            string companyName = "M Inc.";
+            string phone = "777-1111";
+
+            string commandText =
+                $"CREATE TABLE #Shippers{suffix}(" +
+                $"[ShipperID][int] NULL," +
+                $"[CompanyName] [nvarchar] (40) NOT NULL," +
+                $"[Phone] [nvarchar] (24) NULL )" +
+                $"INSERT INTO #Shippers{suffix}" +
+                $"([CompanyName]  " +
+                $",[Phone])" +
+                $"VALUES " +
+                $"('{companyName}' " +
+                $",'{phone}'); " +
+                $"WAITFOR DELAY '0:0:3'; " +
+                $"select s.ShipperID, s.CompanyName, s.Phone " +
+                $"from #Shippers{suffix} s; ";
+            
+            return commandText;
+        }
+
+        [CheckConnStrSetupFact]
+        public static void ExecuteTest()
+        {
+            using (SqlConnection connection = new SqlConnection(DataTestUtility.TcpConnStr))
+            {
+                SqlCommand command = new SqlCommand(GenerateCommandText(), connection);
+                connection.Open();
+
+                IAsyncResult result = command.BeginExecuteReader();
+                while (!result.IsCompleted)
+                {
+                    System.Threading.Thread.Sleep(100);
```

Comment:
This test is supposed to throw `InvalidOperationException` when `The name/value pair "Asynchronous Processing=true" was not included within the connection string defining the connection for this SqlCommand.`
Doc [Link](https://docs.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqlcommand.beginexecutereader?view=netframework-4.7.2#System_Data_SqlClient_SqlCommand_BeginExecuteReader)

Can you check for that condition? If the connection string that is passed in doesn't have Asynchronous Processing set to true, then the test should add the parameter to the connection string. 

Instead of testing BeginExecuteReader with a Sleep statement, could you test an overload which takes in a callback, which is called when the execution finishes? This way you can test EndExecuteReader in the same function and it could provide more branch coverage. 
Alternatively, could you add a test which tests the callback paths?

Reply:
I was confused with "Asynchronous Processing" too. I have added test with callback.